src/app/modules/library/select-all-projects-checkbox/select-all-projects-checkbox.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { MAT_CHECKBOX_DEFAULT_OPTIONS } from '@angular/material/checkbox';
+
+@Component({
+  selector: 'select-all-projects-checkbox',
+  templateUrl: './select-all-projects-checkbox.component.html',
+  providers: [{ provide: MAT_CHECKBOX_DEFAULT_OPTIONS, useValue: { clickAction: 'noop' } }]
+})
+export class SelectAllProjectsCheckboxComponent {
+  @Input() selectedAllProjects: boolean = false;
+  @Input() selectedSomeProjects: boolean = false;
+  @Output() selectAllProjectsEvent: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  protected select(): void {
+    this.selectAllProjectsEvent.emit(!(this.selectedAllProjects || this.selectedSomeProjects));
+  }
```

Comment:
Replace "Project" with "Items"?
```suggestion
  @Input() selectedAllItems: boolean = false;
  @Input() selectedSomeItems: boolean = false;
  @Output() selectAllItemsEvent: EventEmitter<boolean> = new EventEmitter<boolean>();

  protected select(): void {
    this.selectAllItemsEvent.emit(!(this.selectedAllItems || this.selectedSomeItems));
  }
```